scripts/get_test_metrics/get_test_metrics.sh

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env bash
+    
+set -Eeuo pipefail
+trap cleanup SIGINT SIGTERM ERR EXIT
+
+cleanup() {
+  trap - SIGINT SIGTERM ERR EXIT
+  # script cleanup here
+}
+
+script_dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+
+
+usage() {
+      cat <<EOF
+    Usage: $(basename "${BASH_SOURCE[0]}") [-h] [-v] [-p <aws-profile>] -r <aws-region> 
+    [-h]                      : this help message
+    [-v]                      : verbose mode
+    [-p <aws-profile>]        : aws cli profile (optional)
+    -r <aws-region>           : aws region as eu-south-1
+    
+EOF
+  exit 1
+}
+
+parse_params() {
+  # default values of variables set from params
+  project_name=pn
+  work_dir=$HOME/tmp/deploy
+  aws_profile=""
+  aws_region=""
+  env_type=""
+
+
+  while :; do
+    case "${1-}" in
+    -h | --help) usage ;;
+    -v | --verbose) set -x ;;
+    -p | --profile) 
+      aws_profile="${2-}"
+      shift
+      ;;
+    -r | --region) 
+      aws_region="${2-}"
+      shift
+      ;;
+    -w | --work-dir) 
+      work_dir="${2-}"
+      shift
+      ;;
+    -?*) die "Unknown option: $1" ;;
+    *) break ;;
+    esac
+    shift
+  done
+
+  args=("$@")
+
+  # check required params and arguments
+  [[ -z "${aws_region-}" ]] && usage
+  return 0
+}
+
+dump_params(){
+  echo ""
+  echo "######      PARAMETERS      ######"
+  echo "##################################"
+  echo "Project Name:       ${project_name}"
+  echo "Work directory:     ${work_dir}"
+  echo "AWS region:         ${aws_region}"
+  echo "AWS profile:        ${aws_profile}"
+}
+
+# START SCRIPT
+
+parse_params "$@"
+dump_params
+
+
+echo ""
+echo "=== Base AWS command parameters"
+aws_command_base_args=""
+if ( [ ! -z "${aws_profile}" ] ) then
+  aws_command_base_args="${aws_command_base_args} --profile $aws_profile"
+fi
+if ( [ ! -z "${aws_region}" ] ) then
+  aws_command_base_args="${aws_command_base_args} --region  $aws_region"
+fi
+echo ${aws_command_base_args}
+
+echo -n "insert start time: "
+read start_time
+
+echo -n "insert end time: "
+read end_time
+
+
+dir=monitoring_${aws_profile}_$(date +"%d-%m-%Y")
+
+mkdir -p $dir
+
+echo "change work directory"
+
+cd $dir
+
+echo time is $start_time $end_time
+
+echo "start export CPU  metrics for ECS"
+
+echo "==> start export CPU  metrics for ECS"
+
+for cpumetrics in  $(aws ${aws_command_base_args} cloudwatch list-metrics --metric-name CPUUtilization --namespace AWS/ECS  --output text | grep ServiceName | awk '{print $2",Value="$3}') ; do
```

Comment:
@mbrachi sarebbe possibile avere una lista a livello superiore in cui inserisci la lista delle metriche per ogni namespace, in questo caso sarebbe `ECS_Metrics=("CPUUtilization", "MemoryUtilization")` ? così riduciamo un pò il codice e lo rendiamo facilmente modificabile (se voglio aggiungere o togliere una metrica).

Reply:
Aggiunti Array, divisi in base alle statistiche da effettuare (sum or Maximum)